### PR TITLE
LinalgRefactor : Replace elementwise_ops

### PR DESCRIPTION
**Pull request by OXPHOS**

### CODE REVIEW

File: scripts/check_format.sh [hunk 1]
```diff
@@ -30,7 +30,7 @@ function check_shogun_style {
         echo "-----"
         echo "Running clang-format-3.8 against branch ${2:-}, with hash $BASE_COMMIT"
 
-        COMMIT_FILES=$(git diff --name-only $BASE_COMMIT)
+        COMMIT_FILES=$(git diff --name-only $BASE_COMMIT src/shogun)
```
Comment (OXPHOS) [inline scripts/check_format.sh]: Hi @geektoni 
Is there an appropriate way to fix [this error](https://travis-ci.org/shogun-toolbox/shogun/jobs/254065812#L733)?
Or I'll probably try to change the target path to `src/shogun` as I did here.

Comment (geektoni) [inline scripts/check_format.sh]: Hi @OXPHOS
The PR #3934 should solve the issue. Once it gets merged, please rebase this PR on develop and remove the commit https://github.com/shogun-toolbox/shogun/pull/3932/commits/2fc8ab7b002981794bd13e80c4d6d3e3c2b5b020. After that, everything should become green on CIs.

**Comment by vigsterkr** [comment]
@OXPHOS can you rebase as i think @geektoni fixed some of the clang-format errors...

**Comment by OXPHOS** [comment]
Fixed! @vigsterkr @geektoni

**Comment by vigsterkr** [comment]
@OXPHOS merge material?

File: src/shogun/kernel/ExponentialARDKernel.cpp [hunk 2]
```diff
@@ -89,11 +89,7 @@ void CExponentialARDKernel::lazy_update_weights()
 		if (m_ARD_type==KT_SCALAR || m_ARD_type==KT_DIAG)
 		{
 			SGMatrix<float64_t> log_weights(m_log_weights.vector,1,m_log_weights.vlen,false);
-			m_weights_raw=linalg::elementwise_compute(m_log_weights,
-				[  ](float64_t& value)
-				{
-				return CMath::exp(value);
-				});
+			m_weights_raw = linalg::exponent(m_log_weights);
```
Comment (karlnapf) [inline src/shogun/kernel/ExponentialARDKernel.cpp]: yes!

**Pull request merged**